modules/core/server-ts/middleware/website.tsx

```diff
@@ -106,7 +106,13 @@ const renderServerSide = async (req: any, res: any, schema: GraphQLSchema, modul
   context.pageNotFound === true ? res.status(404) : res.status(200);
 
   if (context.url) {
-    res.writeHead(307, { Location: context.url });
+    const redirectBack =
+      context.location && context.location.state && context.location.state.from && context.location.state.from.search
```

Comment:
@mitjade Why do we check `context.location.state.from.search` here?

Reply:
Can we simplify this logic if we will have on client-side: `{ from: props.location.pathname }`, e.g. pass in URL from client-side, not object?

Reply:
If you access `kitUrl/users` it first redirects you to `.../profile` and then to `login`. So I needed to pass the param through multiple redirects.

Reply:
@mitjade Got you, thanks. But I still think passing `{ from: props.location.search || props.location.pathname  }` from client will be less confusing then passing whole object to the server and let server handle it.

Reply:
@vlasenko I took this naming convention from https://reacttraining.com/react-router/web/example/auth-workflow. Will try and just provide a string. Maybe then we could have some naming collisions in this case?

Reply:
@vlasenko It looks like I have to use `state: { from: props.location }`, to be able to access it in `website.tsx`, otherwise it does not work.

Reply:
@mitjade Yes, but you can pass in URL directly from the client-side code, so that you don't need to do processing on the server, e.g. pass from client:
`state: { from: props.location.search || props.location.pathname }`